src/v/storage/batch_cache.cc

```diff
@@ -369,6 +369,22 @@ void batch_cache_index::truncate(model::offset offset) {
     }
 }
 
+void batch_cache_index::prefix_truncate(model::offset offset) {
+    lock_guard lk(*this);
+    if (auto it = _index.lower_bound(offset); it != _index.end()) {
+        if (it == _index.begin()) {
+            return;
+        }
+
+        std::for_each(
+          _index.begin(), std::prev(it), [this](index_type::value_type& e) {
+              _cache->evict(std::move(e.second.range()));
+          });
```

Comment:
existing iteration in the batch cache tends to use the helpers `find_first` and `find_first_contains` which handles some uniqueness of the cache. are there anything in those helpers that would be relevant here?